### ISSUE TITLE
Tests fail on pandas 0.19.1

### ISSUE THREAD

**Issue by zrisher** [open-source-economics/Tax-Calculator#1091]
`test_dropq.test_ascii_output_function()` calls `test_utils.ascii_output()`, which calls: 

```
.to_csv(ascii_filename, header=False, index=False, delim_whitespace=True, sep='\t')
```

on a pandas dataframe. This method doesn't support the `delim_whitespace` keyword arg, but normally doesn't raise an error here since it accepts extra keyword args. This changed in https://github.com/pandas-dev/pandas/pull/13804, so our code raises:

```
TypeError: to_csv() got an unexpected keyword argument 'delim_whitespace'
```

with the newest version of pandas. 

Fortunately, the fix is easy since this keyword argument is superfluous in this context. When reading CSV files, it provides faster parsing when the `sep` argument is ommitted, but it has no meaning when writing since the separator must be explicitly stated.

**Comment by zrisher** [comment]
It also seems these test failures are attributable to pandas 0.18.1:


```
taxcalc\tests\test_dropq.py .....................F.F..........

================================== FAILURES ===================================
___________________________ test_create_json_table ____________________________

    def test_create_json_table():
        df = DataFrame(data=[[1., 2, 3], [4, 5, 6], [7, 8, 9]],
                       columns=['a', 'b', 'c'])
>       ans = create_json_table(df)

taxcalc\tests\test_dropq.py:342:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
taxcalc\dropq\dropq_utils.py:67: in create_json_table
    row_out.append(format_print(df.loc[idx, col], _type, num_decimals))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

x = 2.0, _type = dtype('int64'), num_decimals = 2

    def format_print(x, _type, num_decimals):
        float_types = [float, np.dtype('f8')]
        frmat_str = "0:.{num}f".format(num=num_decimals)
        frmat_str = "{" + frmat_str + "}"
        try:
            if _type in float_types or _type is None:
                return frmat_str.format(x)
            elif _type == int:
                return str(int(x))
            elif _type == str:
                return str(x)
            else:
>               raise NotImplementedError()
E               NotImplementedError

taxcalc\dropq\dropq_utils.py:21: NotImplementedError
______________________ test_format_print_not_implemented ______________________

    def test_format_print_not_implemented():
        x = np.array([1], dtype='i4')
        with pytest.raises(NotImplementedError):
>           format_print(x[0], x.dtype, 2)
E           Failed: DID NOT RAISE <class 'NotImplementedError'>

taxcalc\tests\test_dropq.py:361: Failed
```

It seems as though different data types are attached to frame values by default. I'll look into fixing this Thursday.

**Comment by talumbau** [comment]
@zrisher you are saying that you get these failures with pandas 0.18.1 on Windows? The change in pandas looks like it was slated for version 0.19 and Travis CI currently uses pandas 0.18.1 which passes, but it is Linux. Let me know if the above is really pandas 0.18.1 on Windows.

**Comment by zrisher** [comment]
> Travis CI currently uses pandas 0.18.1 which passes

You're right. The requirement is stated as `pandas 0.18`, and I assumed it was using 0.18.0, because that's what was working before. But looking through the logs it does indeed pass with 0.18.1 on Travis. So perhaps that Pandas version *alone* isn't causing this bug. I will dig into this tonight.

**Comment by zrisher** [comment]
@talumbau I had forgotten that my laptop was set up with a Python 2.7 environment for taxcalc and my desktop was set up with Python 3.4, which turns out to be why `pandas 0.18.0` and `numpy 1.10.4` weren't working for me whereas they had previously - they import and build fine in 2.7 with Conda, but fail with 3.4.
I had actually chosen the two different environments to highlight issues just like these, but I forgot about it until running `conda info` on my laptop.

Now that I have my most recently working env working again, I can see that the above failures in `test_dropq` exist even in my original environment. So they can be disregarded from this issue, and this can be closed as soon as #1092 is merged.

That motivates two more issues though:
* I need to find out why these recently introduced tests fail on Windows but not Linux (something to do with default data types from numpy on different platforms) - tracked in #1099.
* Should we specify the Python version itself within the environment files, and then have 3 files (2.7, 3.4, 3.5)? Then the tests could perhaps refer to those files directly for each environment? @talumbau

**Comment by talumbau** [comment]
> Should we specify the Python version itself within the environment files, and then have 3 files (2.7, 3.4, 3.5)? Then the tests could perhaps refer to those files directly for each environment? @talumbau

For this part, I would say 'no'. The goal would be to have one source code base and one set of dependent packages and versions (e.g. numpy, pandas, etc.) that works on all combinations of {Windows, Mac, Linux} X {Python 2.7, 3.4, 3.5}. This is achievable in nearly all cases without any egregious hacks, and if we end up with a dreaded "if WINDOWS:" flag in some places, that's still better than specifying different versions of numpy for different platforms.


> I need to find out why these recently introduced tests fail on Windows but not Linux (something to do with default data types from numpy on different platforms) - tracked in #1099.

I agree that this seems like the likely candidate. I tried to duplicate on a Windows 7 VM yesterday but my VM hung and I couldn't make progress. I have a Windows 10 VM I can try as well but any efforts here would be greatly appreciated!
